modules/ppcp-button/src/Assets/SmartButton.php

```diff
@@ -231,7 +232,7 @@ public function __construct(
 	 * Registers the necessary action hooks to render the HTML depending on the settings.
 	 *
 	 * @return bool
-	 * @throws \WooCommerce\PayPalCommerce\WcGateway\Exception\NotFoundException When a setting was not found.
+	 * @throws NotFoundException When a setting was not found.
```

Comment:
Removed 54596fc108e2939412ec61f0abc6f072b53c4a01
But PHPstorm still tells me that "PHPDoc comment doesn't contain all the necessary @throws tags"

Reply:
Yeah, it doesn't know about the relation between `has` and `get` (though maybe it can make sense for PHPStorm devs to consider such context for this inspection), and in general technically it's a correct warning because these functions may be independent from each other (but such implementation would not make sense).

But here we intent to avoid exceptions, so adding such `@throws` makes no sense and only confuses those who use these our functions.

A way to avoid the warning is to add `try catch` for these exceptions instead of `if`, but it is less convenient to read/write.

And I mean this issue is in most of the functions with such `@throws`, not just here. You don't have to fix everything now of course, maybe just don't add such `@throws` in the future.

Reply:
ok, understand, thanks.